web/src/containers/ApdPreviousActivityTableTotal.js

```diff
@@ -76,36 +77,9 @@ ApdPreviousActivityTableMMIS.propTypes = {
   totals: PropTypes.object.isRequired
 };
 
-const mapStateToProps = ({
-  apd: {
-    data: { previousActivityExpenses }
-  }
-}) => {
-  const totals = Object.entries(previousActivityExpenses).reduce(
-    (acc, [ffy, expenses]) => ({
-      ...acc,
-      [ffy]: {
-        actual:
-          expenses.hithie.federalActual +
-          [90, 75, 50].reduce(
-            (sum, ffp) => sum + expenses.mmis[ffp].federalActual,
-            0
-          ),
-        approved:
-          expenses.hithie.totalApproved * 0.9 +
-          [90, 75, 50].reduce(
-            (sum, ffp) => sum + (expenses.mmis[ffp].totalApproved * ffp) / 100,
-            0
-          )
-      }
-    }),
-    {}
-  );
-
-  return {
-    totals
-  };
-};
+const mapStateToProps = state => ({
+  totals: selectPreviousActivityExpensesTotals(state)
```

Comment:
Memoized selector.  It does all the math and object-building that `mapStateToProps` used to do, but only when the previous activity expense data changes rather than on every single state change.

Reply:
👏 🌟